This folder is for RNA-seq processing.  
For [normal](https://github.com/STAT540-UBC/team_Bloodies/tree/master/Results/2.RNA-seq/Normal) sample analysis, we did filtering and normalization (see [methods](https://github.com/STAT540-UBC/team_Bloodies/tree/master/Methods/RNAseq_Normal)) and generated [lists](https://github.com/STAT540-UBC/team_Bloodies/tree/master/Results/2.RNA-seq/Normal/Genelist) of differentialy expressed genes. We did sanity check to remove an [outlier]() and checked the cell type [clustering](https://github.com/STAT540-UBC/team_Bloodies/blob/master/Results/2.RNA-seq/Normal/Normal_heatmap_clustering.pdf) based on top2000 abundant genes which showed a group of primitive cells (HSC and MPP) and a group of lymphoid progenitors (MLP, CLP) with GMP intermixed in the lymphoid group.  
We performed full analysis on [leukemia](https://github.com/STAT540-UBC/team_Bloodies/tree/master/Results/2.RNA-seq/Leukemia) samples using limma and generated [p-value distribution histogram](https://github.com/STAT540-UBC/team_Bloodies/blob/master/Results/2.RNA-seq/Leukemia/results/p-value-distribution.pdf) and differential expression [table](https://github.com/STAT540-UBC/team_Bloodies/blob/master/Results/2.RNA-seq/Leukemia/results/toptable.txt).

 